libr/core/panels.c

```diff
@@ -5487,39 +5477,56 @@ void __restore_panel_pos(RPanel* panel) {
 			panel->view->prevPos.w, panel->view->prevPos.h);
 }
 
-char *__get_panels_config_path() {
-	char *configPath = r_str_new (R_JOIN_2_PATHS (R2_HOME_DATADIR, ".r2panels"));
-	if (!configPath) {
-		return NULL;
+char *__get_panels_config_dir_path() {
+	char *config_path = r_str_new (R_JOIN_2_PATHS (R2_HOME_DATADIR, ".r2panels"));
+	char *new_config_path = r_str_home (config_path);
+	R_FREE (config_path);
+	return new_config_path;
+}
+
+char *__create_panels_config_path(const char *file) {
+	char *dir_path = __get_panels_config_dir_path ();
+	r_sys_mkdir (dir_path);
+	char *file_path = r_str_newf (R_JOIN_2_PATHS ("%s", "%s"), dir_path, file);
+	R_FREE (dir_path);
+	return file_path;
+}
+
+char *__get_panels_config_file_from_dir (const char *file) {
+	char *ret = NULL;
+	struct dirent *d;
+	DIR *dir = opendir (__get_panels_config_dir_path ());
+	while ((d = readdir (dir))) {
+		if (!strcmp (d->d_name, file)) {
+			ret = d->d_name;
+		}
 	}
-	char *newPath = r_str_home (configPath);
-	R_FREE (configPath);
-	return newPath;
+	closedir (dir);
+	if (!ret) {
+		return ret;
```

Comment:
do if (ret) { ... 
and put the char dir_path=, ret = ... after that. also the current code is leaking ret